d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -864,31 +974,33 @@ final class RpcRetryTask implements Runnable
     public void run()
     {
       startRpcStreamLocal();
-      for (ResourceType type : _resourceSubscribers.keySet())
+      for (ResourceType originalType : getResourceSubscribers().keySet())
       {
-        Set<String> resources = new HashSet<>(getResourceSubscriberMap(type).keySet());
-        if (resources.isEmpty() && getWildcardResourceSubscriber(type) == null)
+        Set<String> resources = new HashSet<>(getResourceSubscriberMap(originalType).keySet());
+        if (resources.isEmpty())
         {
           continue;
         }
-        ResourceType rewrittenType;
-        if (_subscribeToUriGlobCollection && type == ResourceType.D2_URI_MAP)
+
+        ResourceType adjustedType;
+        if (_subscribeToUriGlobCollection && originalType == ResourceType.D2_URI_MAP)
```

Comment:
The `_subscribeToUriGlobCollection && originalType == ResourceType.D2_URI_MAP` is checked in many places of the code. I think it'd be a little easier to read if it was separated into a standalone function with some docs, like:
```java
private boolean isAdjustedUriGlobCollectionType(ResourceType type) {
  return _subscribeToUriGlobCollection && type == ResourceType.D2_URI_MAP;
}
```

Reply:
updated.